build/src/nbb/build.clj

```diff
@@ -0,0 +1,72 @@
+(ns nbb.build
+  "Provides bb tasks for building and releasing nbb"
+  (:require [babashka.fs :as fs]
+            [babashka.classpath :as classpath]
+            [babashka.tasks :refer [shell clojure]]
+            [clojure.string :as str]
+            [clojure.edn :as edn]))
+
+(defn- feature-files
+  []
+  (filter fs/exists?
+          (map (fn [d]
+                 (fs/file d "nbb_features.edn"))
+               (classpath/split-classpath (classpath/get-classpath)))))
+
+(defn- read-configs
+  [files]
+  (->> files
+       (mapcat (comp edn/read-string slurp str))))
+
+(defn- build-cmd [cmd nbb-dir]
+  (let [files (feature-files)
+        feature-configs (read-configs files)
+        ;; Each ./src/nbb_features.edn has a ./deps.edn
+        feature-dirs (map (comp fs/parent fs/parent) files)
+        cmd' (if (seq files)
+               (format "-Sdeps '%s' %s"
```

Comment:
I tried the `deps/add-deps` and `-Scp (classpath/get-classpath)` approach and got an error `Could not find or load main class clojure.main` that I couldn't get past. This approach works and the constraints seemed ok enough for a first pass so stuck with it

Reply:
Either approach is fine with me